python/cog/server/redis_queue.py

```diff
@@ -192,6 +196,17 @@ def start(self) -> None:
                     for response in self.run_prediction(message, should_cancel):
                         send_response(response)
 
+                    if self.max_failure_count is not None:
+                        # Keep track of runs of failures to catch the situation
+                        # where the worker has gotten into a bad state where it can
+                        # only fail predictions, but isn't exiting.
+                        if response["status"] == Status.FAILED:
+                            failure_count += 1
+                            if failure_count > self.max_failure_count:
+                                self.should_exit = True
```

Comment:
I know we don't have proper telemetry in here yet, but we should probably at least log something so that we can see why the pod stopped and/or search the logs to find out how often this is happening.